lib/global_id/uri/gid.rb

```diff
@@ -83,7 +83,7 @@ def create(app, model, params = nil)
       def build(args)
         parts = Util.make_components_hash(self, args)
         parts[:host] = parts[:app]
-        parts[:path] = "/#{parts[:model_name]}/#{parts[:model_id]}"
+        parts[:path] = "/#{parts[:model_name]}/#{CGI.escape(parts[:model_id].to_s)}"
```

Comment:
@kaspth Look closely, you'll notice I'm calling `to_s` before handing the value to `CGI.escape`, which chokes on, for example, integers.

It's very similar to what goes on in `Object#to_query`:

```ruby
# activesupport/lib/active_support/core_ext/object/to_query.rb:11
def to_query(key)
  "#{CGI.escape(key.to_param)}=#{CGI.escape(to_param.to_s)}"
end
```